### PR TITLE
Create optional otel secret for e2e tests

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

During e2e tests it creates a otel secret that will be used for sending the otel data to provided endpoint.
This is an optional step, so e2e tests will not fail because of missing the secret.

## How can this be tested?

Follow the guide [here](https://github.com/Dynatrace/dynatrace-operator/blob/386af4df2459e81d888c54650184da444b82fa66/doc/otel.md#L1-L1), there was an extra section added.

## Checklist

- ~[ ] Unit tests have been updated/added~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: test/helpers/tenant/secrets.go [hunk 1]
```diff
@@ -15,26 +15,30 @@ import (
...
 	Tenants []Secret `yaml:"tenants"`
 }
 
 type Secret struct {
-	TenantUid                       string `yaml:"tenantUid"`
-	ApiUrl                          string `yaml:"apiUrl"`
-	ApiToken                        string `yaml:"apiToken"`
-	SyntheticLocEntityId            string `yaml:"syntheticLocEntityId"`
-	SyntheticBrowserMonitorEntityId string `yaml:"syntheticBrowserMonitorEntityId"`
+	TenantUid string `yaml:"tenantUid"`
```
Comment (waodim) [inline test/helpers/tenant/secrets.go:39]: Am I wrong or isn't it called TenantUUID?

Comment (0sewa0) [reply test/helpers/tenant/secrets.go:39]: I didn't name that, I just removed/moved that part 😬

File: cmd/csi/provisioner/builder.go [hunk 2]
```diff
@@ -1,8 +1,6 @@
 package provisioner
```
Comment (waodim) [inline cmd/csi/provisioner/builder.go:1]: Apart from the failing unit tests I would approve it as it lgtm.

**Review by waodim — APPROVED** [review]
tested it - secret is there if provided. code lgtm

**Pull request merged**